conf/modules.config

```diff
@@ -1,153 +1,70 @@
-        ext.prefix = File name prefix for output files.
-----------------------------------------------------------------------------------------
-*/
-
 process {
 
-    publishDir = [
-        path: { "${params.outdir}/${task.process.tokenize(':')[-1].tokenize('_')[0].toLowerCase()}" },
-        mode: params.publish_dir_mode,
-        saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
-    ]
     withName: METADATA_TO_SAMPLESHEET {
         publishDir = [
-            path:"${params.outdir}",
-            mode: params.publish_dir_mode        ]
-    }
-    withName: BWAMEM2_MEM {
-        ext.prefix = { "${meta.id}" }
-        publishDir = [
-            enabled: false
-        ]
-    }
-    withName: SAVE_REFERENCE {
-        publishDir = [
-            path: params.save_reference_path ? params.save_reference_path : params.outdir,
-            mode: params.publish_dir_mode,
-            pattern: "reference/*"
-        ]
-    }
-    withName: BWAMEM2_INDEX {
-        ext.prefix = { "${meta.id}" }
-        publishDir = [
-            enabled: false
-        ]
-    }
-    withName: SAMTOOLS_FAIDX {
-        ext.prefix = { "${meta.id}" }
-        publishDir = [
-            enabled: false
-        ]
-    }
-    withName: SAMTOOLS_SORT {
-        ext.prefix = { "${meta.id}.sorted" }
-        publishDir = [
-            enabled: false
-        ]
-    }
-    withName: SAMTOOLS_INDEX {
-        ext.prefix = { "${meta.id}" }
-        publishDir = [
-            enabled: false
-        ]
-    }
-    withName: SAMTOOLS_FLAGSTAT {
-        ext.prefix = { "${meta.id}" }
-        publishDir = [
-            enabled: false
-        ]
-    }
-    withName: SAMTOOLS_IDXSTATS {
-        ext.prefix = { "${meta.id}" }
-        publishDir = [
-            enabled: false
-        ]
-    }
-    withName: FASTQC {
-        ext.args = '--quiet'
-        publishDir = [
-            enabled: false
-        ]
-    }
-    withName: CAT_FASTQ {
-        publishDir = [
-            enabled: false
-        ]
-    }
-    withName: FASTP {
-        publishDir = [
-            enabled: false
+            path: "${params.outdir}",
+            mode: params.publish_dir_mode
         ]
     }
-    withName: MULTIQC {
-        ext.args   = { params.multiqc_title ? "--title \"$params.multiqc_title\"" : '' }
+
+    withName: 'GRZQC:PREPARE_REFERENCES:(SAMTOOLS_BGZIP|BWAMEM2_INDEX)' {
         publishDir = [
-            path: { "${params.outdir}/multiqc" },
+            path: { "${params.reference_path}/${meta.id}" },
             mode: params.publish_dir_mode,
+            enabled: "${params.reference_path != null}",
             saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
         ]
     }
+
     withName: PICARD_ADDORREPLACEREADGROUPS {
-        scratch = true
-        ext.args =  { [
-            "--VALIDATION_STRINGENCY LENIENT",
-            "--RGLB lib",
```

Comment:
I would prefer to change those params if it is not related to this PR. To be able to keep things tracable. scratch = true is implemented by Yun to resolve tmp_dir issue for example.